src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -93,12 +122,46 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     }
 
     async deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._deinitializePollingTimer();
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private _initializePollingMechanism(submitForm: () => void, gatewayId?: string) {
+        if (gatewayId === PaymentStrategyType.PAYPAL_COMMERCE_ALTERNATIVE_METHODS) {
+            this._pollingInterval = window.setTimeout(async () => {
+                try {
+                    this._pollingTimer += POLLING_INTERVAL;
+                    const res = await this._paypalCommerceRequestSender.getOrderStatus();
+                    if (res.status !== ORDER_STATUS_CREATED && res.status !== ORDER_STATUS_APPROVED) {
+                        this._deinitializePollingTimer();
+                    }
+                    if (res.status === ORDER_STATUS_APPROVED) {
+                        this._deinitializePollingTimer();
+                        this._tokenizePayment({ orderID: this._paypalCommercePaymentProcessor.getOrderId() }, submitForm);
+                    } else {
+                        if (this._pollingTimer < POLLING_MAX_TIME) {
+                            this._initializePollingMechanism(submitForm, gatewayId);
+                        } else {
+                            this._deinitializePollingTimer();
+
+                            return;
+                        }
+                    }
+                } catch (e) {
+                    this._deinitializePollingTimer();
```

Comment:
I can see that the polling stops after 10min. But it also stops if there's any error within 10min. (i.e.: network error). Is this intentional? Do you think it's a good idea to retry for a few times instead? Also, do you want log the error somehow instead of swallowing it entirely? i.e.: You can add an `onError` callback so that the client app (UCO) can hook into it and report the error in Sentry?

Reply:
Thanks for the question, at the moment we are checking the way to stop the payment session entirely and show an error. 
Ideally, at the end of the timer, we want to close the PayPal popup and overlay and throw an error, it is possible to achieve via `paypal.close()` of the PayPal SDK method. However, in addition to the PayPal popup shut down, the iframe buttons will be deinitialized as well, hence we need to re-initialize them as well. We decided to move further with the testing of what we have now to make sure the poller works as expected and continue on the improvement with graceful finalization of the "timer finalized" scenario.